packages/core/src/util.js

```diff
@@ -0,0 +1,34 @@
+export const containsAllParents = (leaf, parents) => {
+  if (parents.length === 0) {
+    return true;
+  }
+  let containsAll = true;
+  parents.forEach((parent, index) => {
+    if (parent !== leaf[index]) {
+      containsAll = false;
+    }
+  });
+  return containsAll;
+};
+
+export const filterBySearchTerm = (leaf, searchTerm) => {
+  return (
+    searchTerm === "" ||
+    leaf[leaf.length - 1].toLowerCase().includes(searchTerm.toLowerCase())
+  );
+};
+
+export const buildLeafForItemRenderer = (leaf, currentDepth, searchTerm) => {
+  return {
+    item: leaf,
+    hasChild: leaf.length - 1 > currentDepth && searchTerm === "",
+    currentDepth,
+    key: searchTerm === "" ? leaf[currentDepth] : leaf.slice().toString()
+  };
+};
+
+export const removeDuplicateLeafs = leaves => {
```

Comment:
Are we generating duplicates? Is there any reason we'll get duplicates?

Reply:
this is the way I display one parent of a group of leaves